cmd/web.go

```diff
@@ -577,6 +581,18 @@ func runWeb(ctx *cli.Context) {
 		err = server.ListenAndServeTLS(setting.CertFile, setting.KeyFile)
 	case setting.FCGI:
 		err = fcgi.Serve(nil, m)
+	case setting.UNIX_SOCKET:
+		os.Remove(listenAddr)
+		listener, err := net.Listen("unix", listenAddr)
+		defer listener.Close()
```

Comment:
You are right, this needs to be moved after the error check. The Go source for `net.UnixListen` is has this line in it:

```
if err != nil {
	return nil, &OpError{Op: "listen", Net: net, Source: nil, Addr: laddr.opAddr(), Err: err}
}
```